### ISSUE TITLE
The repo "https://github.com/netbox-community/devicetype-library.git" is not a valid git repo.

### ISSUE THREAD

**Issue by raad-orfali** [minitriga/Netbox-Device-Type-Library-Import#129]
I Followed the instruction to clone, config .env and run the ./nb-dt-import.py but getting this error.

Note that I don't have net issue and I can git clone https://github.com/netbox-community/devicetype-library.git without any issues

I also tried to create empty repo folder manually and run the command but getting the below error:

Package devicetype-library is already installed, updating /opt/Device-Type-Library-Import/repo
An unknown error occurred: "Git Repository Error"

**Comment by raad-orfali** [comment]
Thanks a lot problem solved.

kindly update the README if possible to reflect the same for new users.

**Comment by mhdan** [comment]
I have the same problem but removing the GitInvalidRepositoryError line does not solve my problem.
any idea what is the main reason of this issue?

**Comment by Azmodeszer** [comment]
Same here...

**Comment by Azmodeszer** [comment]
Any news?

**Comment by mckechniefraz** [comment]
I was able to fix the issue by commenting the below code in repo.py

`            # if not self.repo.remotes.origin.url.endswith('.git'):
            #     self.handle.exception("GitInvalidRepositoryError", self.repo.remotes.origin.url,
            #                           f"Origin URL {self.repo.remotes.origin.url} does not end with .git")`

**Comment by Azmodeszer** [comment]
Yes, that was suggested earlier in this issue, but removing it does not work for me.

**Comment by mhdan** [comment]
In my case the problem was that cloning the repo does take lots of time and the main reason was it.
I solved the problem by manually downloading the repo and unzipping it to the required path. 

The timeout is caught and prints the wrong error and it confuses people.

**Comment by Azmodeszer** [comment]
Still getting `The repo "https://github.com/netbox-community/devicetype-library" is not a valid git repo.`

**Comment by DrDoc777** [comment]
> Still getting `The repo "https://github.com/netbox-community/devicetype-library" is not a valid git repo.`

Had same here, but after installed on a fresh ubuntu "as-is" I don't have the problem on the fresh one; even if the https://github.com/netbox-community/devicetype-library.git is specified in the .env file

**Comment by JarrettStidham** [comment]
change the repo folder， Modify permissions。 
sudo chmod -R 777 /opt/Device-Type-Library-Import/venv
sudo chmod -R a+w /opt/Device-Type-Library-Import/repo
./nb-dt-import.py

**Comment by jzrawlins** [comment]
Getting no such file.
